### PR TITLE
Make is_multilabel an attribute of Dataset

### PR DESCRIPTION
**Pull request by achyudh**
### Changes:
- Make is_multilabel an attribute of Dataset
- Rename ReutersEvaluator to ClassificationEvaluator
- Rename ReutersTrainer to ClassificationTrainer
- Remove SST dataset
- Move save_path to global args file to address issue #2

### CODE REVIEW

**Comment by achyudh** [comment]
This pull request has been verified working for kim_cnn and lstm_reg, but breaks the other models. We need to make the same changes to the driver files of other models to get them working.

File: common/evaluators/classification_evaluator.py [hunk 1]
```diff
@@ -37,15 +36,15 @@ def get_scores(self):
...
-                target_labels.extend(torch.argmax(batch.label, dim=1).cpu().detach().numpy())
-                total_loss += F.cross_entropy(scores, torch.argmax(batch.label, dim=1), size_average=False).item()
-            else:
+            if self.is_multilabel:
                 scores_rounded = F.sigmoid(scores).round().long()
                 predicted_labels.extend(scores_rounded.cpu().detach().numpy())
                 target_labels.extend(batch.label.cpu().detach().numpy())
                 total_loss += F.binary_cross_entropy_with_logits(scores, batch.label.float(), size_average=False).item()
+            else:
+                predicted_labels.extend(torch.argmax(scores, dim=1).cpu().detach().numpy())
```
Comment (daemon) [inline common/evaluators/classification_evaluator.py:45]: Is the `detach()` still necessary?

File: common/trainers/classification_trainer.py [hunk 2]
```diff
@@ -10,10 +10,10 @@
 from .trainer import Trainer
 
 
-class ReutersTrainer(Trainer):
+class ClassificationTrainer(Trainer):
 
     def __init__(self, model, embedding, train_loader, trainer_config, train_evaluator, test_evaluator, dev_evaluator):
-        super(ReutersTrainer, self).__init__(model, embedding, train_loader, trainer_config, train_evaluator, test_evaluator, dev_evaluator)
+        super(ClassificationTrainer, self).__init__(model, embedding, train_loader, trainer_config, train_evaluator, test_evaluator, dev_evaluator)
```
Comment (daemon) [inline common/trainers/classification_trainer.py]: `super().__init__(...)` is sufficient in Python 3.

File: common/trainers/classification_trainer.py [hunk 3]
```diff
@@ -44,18 +44,17 @@ def train_epoch(self, epoch):
                 else:
                     scores = self.model(batch.text[0], lengths=batch.text[1])
 
-            if 'single_label' in self.config and self.config['single_label']:
-                for tensor1, tensor2 in zip(torch.argmax(scores, dim=1), torch.argmax(batch.label.data, dim=1)):
-                    if np.array_equal(tensor1, tensor2):
-                        n_correct += 1
-                loss = F.cross_entropy(scores, torch.argmax(batch.label.data, dim=1))
-            else:
+            if 'is_multilabel' in self.config and self.config['is_multilabel']:
```
Comment (daemon) [inline common/trainers/classification_trainer.py:48]: `if self.config.get('is_multilabel')`?

File: models/char_cnn/args.py [hunk 4]
```diff
@@ -1,34 +1,24 @@
...
-    parser.add_argument('--word-vectors-dir', help='word vectors directory',
-                        default=os.path.join(os.pardir, 'Castor-data', 'embeddings', 'word2vec'))
-    parser.add_argument('--word-vectors-file', help='word vectors filename', default='GoogleNews-vectors-negative300.txt')
-    parser.add_argument('--trained-model', type=str, default="")
     parser.add_argument('--weight-decay', type=float, default=0)
 
+    parser.add_argument('--word-vectors-dir', default=os.path.join(os.pardir, 'Castor-data', 'embeddings', 'word2vec'))
+    parser.add_argument('--word-vectors-file', default='GoogleNews-vectors-negative300.txt')
+    parser.add_argument('--save-path', type=str, default=os.path.join('char_cnn', 'saves'))
+    parser.add_argument('--resume-snapshot', type=str, default=None)
```
Comment (daemon) [inline models/char_cnn/args.py]: I'm not sure if `default=None` is necessary.

**Review by daemon — COMMENTED** [review]
Minor comments.

File: common/trainers/classification_trainer.py (hunk 3 above)
Comment (Ashutosh-Adhikari) [reply common/trainers/classification_trainer.py:48]: this is same as the hasattr() test that we talked about. Is there an elegant workaround for this?

File: common/evaluate.py [hunk 5]
```diff
@@ -1,18 +1,15 @@
...
+from common.evaluators.classification_evaluator import ClassificationEvaluator
 
 
 class EvaluatorFactory(object):
     """
     Get the corresponding Evaluator class for a particular dataset.
     """
     evaluator_map = {
-        'SST-1': SSTEvaluator,
-        'SST-2': SSTEvaluator,
```
Comment (Ashutosh-Adhikari) [inline common/evaluate.py:11]: So we have decided to remove SST's altogether?

File: common/trainers/classification_trainer.py (hunk 3 above)
Comment (achyudh) [reply common/trainers/classification_trainer.py:48]: No, this is different from the one we talked about. We talked about TAR attribute. 

But this also needs to be fixed with the get() method with a default return value in case the key isn't present.

File: common/evaluate.py (hunk 5 above)
Comment (Ashutosh-Adhikari) [reply common/evaluate.py:11]: @achyudh given that the current changes aren't causing any performance change, shall I merge the PR? CC: @daemon 

I find this PR okay!

Comment (achyudh) [reply common/evaluate.py:11]: Yeah, we decided to remove SST and add it back in later. I checked for performance changes, didn't find any when I ran LSTM Reg and KimCNN for 3 epochs.

File: models/char_cnn/args.py (hunk 4 above)
Comment (achyudh) [reply models/char_cnn/args.py]: I agree. I think the same applies to `parser.add_argument('--trained-model', type=str)`

File: common/trainers/classification_trainer.py (hunk 2 above)
Comment (achyudh) [reply common/trainers/classification_trainer.py]: Agreed. Fixed in ecda4c0409cc5f219eac0873e42b79de79f79861

**Comment by achyudh** [comment]
Thanks @Ashutosh-Adhikari and @daemon for the review. Merged into master.
